app/factories/shopify/variant_factory.rb

```diff
@@ -0,0 +1,31 @@
+module Shopify
+  class VariantFactory
```

Comment:
This doesn't seem like a Factory (which, conventionally, creates new objects) rather than a Converter or Migrator or Copier. It encapsulates the knowledge of how to mutate a Shopify variant into a Spree variant.